src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/HttpClientTransport.cs

```diff
@@ -15,16 +15,10 @@ namespace Azure.Base.Http.Pipeline
 {
     public class HttpClientTransport : HttpPipelineTransport
     {
-        static readonly HttpClient s_defaultClient = new HttpClient();
+        public Action<HttpMessageHandler> ConfigureHandler { get; set; }
+        public HttpClient Client { get; set; }
 
-        readonly HttpClient _client;
-
-        public HttpClientTransport(HttpClient client = null)
-            => _client = client == null ? s_defaultClient : client;
-
-        public readonly static HttpClientTransport Shared = new HttpClientTransport();
-
-        public sealed override HttpMessage CreateMessage(IServiceProvider services, CancellationToken cancellation)
```

Comment:
Why did you remove IServiceProvider?

Reply:
Presumably, you need IServiceProvider in case a policy needs to revolve types from DI. But in that case you would prefer the policy type to be resolved from DI itself which new design allows to do (the factory delegate would call into DI) so there is no need to put IServiceProvider on message.

Reply:
We don't have an end-to-end DI solution in the platform. People new up policies, people want to use services without having to change APIs, and many other problems.

Reply:
There is no need for IServiceProvider in current implementation at all. Why complicate things?